design/storage-class-device-set.md

```diff
@@ -0,0 +1,69 @@
+# Rook StorageClassDeviceSets
+
+**Target version**: Rook 1.1
+
+## Background
+The primary motivation for this feature is to take advantage of the mobility of storage in cloud-based environments by defining storage based on sets of devices consumed as PVCs. In environments like AWS you can request remote storage that can be dynamically provisioned and attached to any node in a given Availability Zone (AZ). The design can also accommodate non-cloud environments through the use of local PVs.
+
+## StorageClassDeviceSet struct
+```go
+// File: pkg/apis/ceph.rook.io/v1/types.go
+package v1
+
+import (
+    v1 "k8s.io/api/core/v1"
+
+    rook "github.com/rook/rook/pkg/apis/rook.io/v1alpha2"
+)
+
+type StorageClassDeviceSets []StorageClassDeviceSet
+
+struct StorageClassDeviceSet {
+    Name                 string // A unique identifier for the set
+    Count                int // Number of devices in this set
+    Resources            v1.ResourceRequirements // Requests/limits for the devices
```

Comment:
Does this mean QoS?

Reply:
It means CPU and memory requests/limits. Kubernetes will use this to 1.) determine if a Pod can be scheduled on the node based on available CPU and memory on the node, and 2.) determine priority when a node starts running out of resources about which pods to start evicting.